### PR TITLE
[HEXDEV-717] enable SPNEGO authentication

### PR DESCRIPTION
**Pull request by honzasterba**
back end part done, working on Python API support

### CODE REVIEW

File: h2o-jetty-9/src/main/java/water/webserver/jetty9/Jetty9Helper.java [hunk 1]
```diff
@@ -1,15 +1,10 @@
...
 
 import org.eclipse.jetty.jaas.JAASLoginService;
-import org.eclipse.jetty.security.Authenticator;
-import org.eclipse.jetty.security.ConstraintMapping;
-import org.eclipse.jetty.security.ConstraintSecurityHandler;
-import org.eclipse.jetty.security.DefaultIdentityService;
-import org.eclipse.jetty.security.HashLoginService;
-import org.eclipse.jetty.security.IdentityService;
-import org.eclipse.jetty.security.LoginService;
+import org.eclipse.jetty.security.*;
```
Comment (pkozelka) [inline h2o-jetty-9/src/main/java/water/webserver/jetty9/Jetty9Helper.java]: pls keep imports explicit

**Comment by honzasterba** [comment]
now how do hell do we test this? :)

File: h2o-py/requirements.txt [hunk 2]
```diff
@@ -2,6 +2,8 @@ pip
 colorama>=0.3.8
 future
 requests
+winkerberos >= 0.5.0; sys.platform == 'win32'
```
Comment (michalkurka) [inline h2o-py/requirements.txt]: Just please keep in mind It shouldn't be a hard requirement, only an optional one.

File: h2o-py/h2o/auth.py [hunk 3]
```diff
@@ -0,0 +1,60 @@
+# -*- encoding: utf-8 -*-
+"""
+Data class for SPNEGO authentication.
+
+:copyright: (c) 2016 H2O.ai
+:license:   Apache License Version 2.0 (see LICENSE for details)
+"""
+from __future__ import absolute_import, division, print_function, unicode_literals
+
+try:
```
Comment (michalkurka) [inline h2o-py/h2o/auth.py]: kerberos should be optional, we need to make sure the code doesn't fail if kerberos is not installed

and produce an user-understandable error message

**Comment by michalkurka** [comment]
As perfect as anything kerberos related can be ;)

File: h2o-py/h2o/auth.py (hunk 3 above)
Comment (honzasterba) [reply h2o-py/h2o/auth.py]: added a nicer message, this fails only when user tries to explicitly import from h2.auth, does not influence any other imports

**Comment by honzasterba** [comment]
@michalkurka should be ready to merge, I would like to do the tests in another PR

**Pull request merged**